afl binary cannot be built

after "cargo afl build", there is an error.

`   Compiling png-afl v0.2.0 (/home/kym000131/fuzzing_set/Rust/image-png/png-afl)
error: declaration of a `no_mangle` function
  --> src/main.rs:11:1
   |
11 | #[no_mangle]
   | ^^^^^^^^^^^^
   |
   = note: the linker's behavior with multiple libraries exporting duplicate symbol names is undefined and Rust cannot provide guarantees when you manually override them
note: the lint level is defined here
  --> src/main.rs:2:11
   |
2  | #![forbid(unsafe_code)]
   |           ^^^^^^^^^^^

error[E0308]: mismatched types
  --> src/main.rs:19:9
   |
19 |     let (info, mut reader) = decoder.read_info().map_err(|_| ())?;
   |         ^^^^^^^^^^^^^^^^^^   ------------------------------------ this expression has type `Reader<&[u8]>`
   |         |
   |         expected `Reader<&[u8]>`, found `(_, _)`
   |
   = note: expected struct `Reader<&[u8]>`
               found tuple `(_, _)``
              
I think there is a problem in png-afl/src/main.rs

--- a/png-afl/src/main.rs
+++ b/png-afl/src/main.rs
@@ -1,17 +1,8 @@
-
 #![forbid(unsafe_code)]
 
 #[macro_use]
 extern crate afl;
 extern crate png;
-
-// detect_odr_violation=0 is for https://github.com/rust-lang/rust/issues/41807
-const ASAN_DEFAULT_OPTIONS: &'static [u8] = b"detect_odr_violation=0\0";
-
-#[no_mangle]
-pub extern "C" fn __asan_default_options() -> *const u8 {
-    ASAN_DEFAULT_OPTIONS as *const [u8] as *const u8
-}
 
 #[inline(always)]
 fn png_decode(data: &[u8]) -> Result<(png::OutputInfo, Vec<u8>), ()> {

--- a/png-afl/src/main.rs
+++ b/png-afl/src/main.rs
@@ -7,14 +7,16 @@ extern crate png;
 #[inline(always)]
 fn png_decode(data: &[u8]) -> Result<(png::OutputInfo, Vec<u8>), ()> {
     let decoder = png::Decoder::new(data);
-    let (info, mut reader) = decoder.read_info().map_err(|_| ())?;
 
-    if info.buffer_size() > 5_000_000 {
+    let mut reader = decoder.read_info().map_err(|_| ())?;
+
+    let buffer_size = reader.output_buffer_size();
+    if buffer_size > 5_000_000 {
         return Err(());
     }
 
-    let mut img_data = vec![0u8; info.buffer_size()];
-    reader.next_frame(&mut img_data).map_err(|_| ())?;
+    let mut img_data = vec![0u8; buffer_size];
+    let info = reader.next_frame(&mut img_data).map_err(|_| ())?;
 
     Ok((info, img_data))
 }

